test/MLJ/unit.jl

```diff
@@ -12,6 +12,13 @@ struct DummyModel
 end
 DummyModel(;kwargs...) = DummyModel(true) # we literally don't care,
                                           # we just need a constructor
+mutable struct DummyGenericKwargModel{T}
```

Comment:
I added a dummy model mimicking a model with a generic kwarg constructor (as mentioned in the comments). I didn't want to bring additional dependencies to tests. For the reviewer, this test will fail on the existing version of TreeParzen due to the same MethodError for unsupported argument as raised in the issue.